### PR TITLE
Uploan unleash client implementation.

### PR DESCRIPTION
**Pull request by ryanermita**
## Overview
Solves 
- [Secure Unleash API Client](https://uploan.atlassian.net/browse/ISVC-65)
- [Secure Unleash Admin Client](https://uploan.atlassian.net/browse/ISVC-66)

This pull request contains the implementation of Uploan with Unleash Client as a Feature Flag Back-end Service.

I have:

- [x] Tested this locally and reviewed my own code.
- [ ] Provided appropriate unit tests for my change.
- [x] Added a description of my changes.
- [ ] Provided test instructions for the reviewer.

## Analysis
Uploan will use unleash client as the back-end application of its feature flag service. With this goal, Unleash client should be implemented with Uploan Authentication Service for authentication and authorization workflow. This client service will be an internal tool within Uploan thus, the implementation of Unleash-frontend with Uploan's microservice ecosystem.

## Guides
For the development standards, coding best practices, and code review guidelines. Kindly refer to this document: [Uploan Engineering Handbooks](https://uploan.atlassian.net/wiki/spaces/0T/pages/662340021/Handbooks)

### CODE REVIEW

File: uploan/uploan-auth-hook.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    });
+
+    app.use('/api/admin/', async (req, res, next) => {
+        const isValidToken = await uploanAuth.isValidAuthToken(req)
+        if (!isValidToken.result) {
+            return res
+                .status('401')
+                .json(
+                    new AuthenticationRequired({
+                        path: 'https://gw.dev.uploan.ph/api/v1/login',
```
Comment (lcmandrada-uploan) [inline uploan/uploan-auth-hook.js:23]: Should `path` be a variable or will it always use the gateway in DEV environment?

File: uploan/uploan-auth-hook.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                        type: 'uploan',
+                        message:
+                            'You have to indentify yourself in order to use Unleash.',
```
Comment (lcmandrada-uploan) [inline uploan/uploan-auth-hook.js:26]: Typo in `indentify`

Comment (ryanermita) [reply uploan/uploan-auth-hook.js:26]: good catch!

File: uploan/uploan-auth-hook.js (hunk 1 above)
Comment (ryanermita) [reply uploan/uploan-auth-hook.js:23]: good catch!

**Pull request closed without merging**